client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js

```diff
@@ -90,26 +93,26 @@ angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
       Until the task is finished, the diagnostic we get from
       the API is IRRELEVANT and should not be displayed to the user.
     */
-    this.$rootScope.$on('accessDiagnosticDetails:launch', () => {
-      this.launchDiagnostic();
-    });
+    this.$rootScope.$on('accessDiagnosticDetails:launch', () => this.launchDiagnostic());
 
     this.$rootScope.$on('accessDiagnosticDetails:get', () => {
+      let launchDiagnostic;
       if (!this.accessDiagnostic && !this.$scope.loaders.accessDiagnosticLaunched) {
-        this.launchDiagnostic();
+        launchDiagnostic = this.launchDiagnostic();
       }
       this.$rootScope.$broadcast('accessDiagnosticDetails:arrived', this.accessDiagnostic);
+      return launchDiagnostic;
```

Comment:
Hmm `$on` returns a function to unregister the event, so this wouldn't have any effect